cpp/src/arrow/filesystem/test_util.h

```diff
@@ -74,6 +74,10 @@ void CreateFile(FileSystem* fs, const std::string& path, const std::string& data
 ARROW_TESTING_EXPORT
 void SortInfos(FileInfoVector* infos);
 
+// Create a copy of a FileInfo vector sorted by lexicographic path order
+ARROW_TESTING_EXPORT
+std::vector<FileInfo> SortedInfos(const std::vector<FileInfo>& infos);
```

Comment:
Nit, but nice addition anyway
```suggestion
FileInfoVector SortedInfos(const FileInfoVector& infos);
```